//Lembre-se que a mensagem do erro aparece apenas quando o teste falha . Experimente alterar os parÃ¢metros de entrada da funÃ§Ã£o add e veja o resultado!
const assert = require('assert');

function add(a, b) {
  return a + b;
}

const expected = add(1, 2);

assert.ok(expected === 3, 'Um mais dois Ã© igual a trÃªs'); // Checa se o primeiro argumento Ã© verdadeiro
assert.strictEqual(expected, 3, 'Um mais dois Ã© igual ao NÃšMERO trÃªs'); // Checa se o primeiro e segundo argumentos sÃ£o iguais em valor e tipo (===)
assert.notStrictEqual(expected, 4, 'Um mais dois Ã© igual a trÃªs (e nÃ£o quatro!)'); // Checa se o primeiro e segundo argumentos sÃ£o diferentes (!==) 

// Ã‰ possÃ­vel tambÃ©m testar estruturas como arrays e objetos (assert jÃ¡ declarado lÃ¡ em cima):

const list1 = [1, 2, 3, 4, 5];
const list2 = [1, 2, 3, 4, 5];

assert.deepStrictEqual(list1, list2, 'Erro: list1 e list2 nÃ£o sÃ£o estritamente iguais');

const person1 = { name: 'john', age: 21 };
const person2 = { name: 'john', age: 21 };

assert.deepStrictEqual(person1, person2, 'Erro: person1 e person2 nÃ£o sÃ£o estritamente iguais');

const person3 = { name: 'john', age: 19 };

assert.notDeepStrictEqual(person1, person3, 'Erro: os valores dos objetos sÃ£o estritamente iguais');

// Um papel dos testes unitÃ¡rios Ã© nos ajudar a implementar funÃ§Ãµes mais robustas. Veja, por exemplo, a funÃ§Ã£o que realiza a divisÃ£o:

function division(x, y) {
  return x / y;
}

// Vamos aos testes: 

assert.strictEqual(division(10, 2), 5); // OK
assert.strictEqual(division(10, 0), 0); // ðŸ’£

// O primeiro assert.strictEqual funciona e nÃ£o apresenta erro, mas o segundo deixa evidente uma fragilidade na implementaÃ§Ã£o da funÃ§Ã£o division , pois nÃ£o poderia ser possÃ­vel realizar a divisÃ£o por 0. Nesse caso, devemos melhorar a implementaÃ§Ã£o da funÃ§Ã£o division , a fim de nÃ£o termos mais esse estado inconsistente.

function divisionFixed(x, y) {
  if (y === 0) throw new Error('parameter y must not be 0');
  return x / y;
}

assert.strictEqual(divisionFixed(10, 2), 5); // OK
assert.throws(() => { divisionFixed(10, 0); }, 'Error: parameter y must not be 0'); // OK